spec/unit/indirector/file_server_spec.rb

```diff
@@ -206,6 +206,28 @@ module Testing; end
       @file_server.search(@request)
     end
 
+    describe "should set 'checksum_type' on the instances if it is set in the request options" do
+      [:md5, :md5lite, :sha256, :sha256lite, :mtime, :ctime].each do |checksum_type|
+        it "to #{checksum_type}" do
+          @configuration.expects(:split_path).with(@request).returns([@mount, "rel/path"])
+
+          @mount.expects(:search).with { |key, request| key == "rel/path" }.returns []
+
+          Puppet::FileSystem.stubs(:exist?).returns true
```

Comment:
This was pulled from file_content/file_server_spec.rb. I'll take a look at whether both can be cleaned up.